gapic/src/main/com/google/gapid/views/AtomTree.java

```diff
@@ -450,7 +455,13 @@ public void updateElement(Object parent, int index) {
       AtomStream.Node child = ((AtomStream.Node)parent).getChild(index);
       atoms.load(child, refresher::refresh);
       viewer.replace(parent, index, child);
-      viewer.setChildCount(child, child.getChildCount());
+      if (child.getChildCount() > 100) {
+        // For nodes with lots of children, only allocate the children, once the node
+        // is actually expanded.
+        viewer.setHasChildren(child, true);
```

Comment:
Sorry, I didn't make my original question that clear. I'm curious to know if we can kill the `if (child.getChildCount() > 100)` conditional and just replace it with `viewer.setHasChildren(child, true);`